### ISSUE TITLE
[data grid] Support onFilterMenuClose

### ISSUE THREAD

**Issue by Janpot** [mui/mui-x#11335]
### Summary 💡

The data grid should support a `onFilterModelChange` or any other mechanism that allows delaying the filter update until the user has decided their filter is finished. 

### Motivation 🔦

This is useful in server-side filtering scenarios. The current behavior only allows for immediate input with a debounce (if I understand correctly). But that leads to overfetching.

**Search keywords**: onFilterModelChange close

**Comment by Janpot** [comment]
>  Let me know if you meant something different.

Yes, it fires every time the model changes, with a debounce. That's ok for client side filtering. But for serverside filtering, it will refetch too often. I'd like to fire off the request only once the user has closed the filter menu.

**Comment by DanailH** [comment]
Got it. I guess we can add a filter panel close prop/event. Actually, if we fix the `onMenuClose` callback to behave as expected that can be a solution but it should pass which menu is being closed.

**Comment by Janpot** [comment]
> Another option would be to alow for fine-tuning the debounce timer.

You already can, but that wouldn't really help. I want to fire of the request as soon as the menu closes, regardless if there's still debounce time left.

**Comment by michelengelen** [comment]
Thanks for the request. This is now possible. #23023 added [`<GridFilterPanelBase />`](https://mui.com/x/api/data-grid/grid-filter-panel-base/), a controlled filter panel released in [v9.9.0](https://github.com/mui/mui-x/releases/tag/v9.9.0). Keep a draft filter model in your own state and sync it to the grid on demand. See the [Apply filters on demand](https://mui.com/x/react-data-grid/filtering/customization/#apply-filters-on-demand) recipe. To apply on close instead of an Apply button, commit the draft when the panel unmounts, or subscribe to the `preferencePanelClose` event. This avoids a server request on every change. The same request was tracked in #1723, closed as completed. I will close this issue as resolved. Comment here if your use case is not covered.
